### DISCUSSION TITLE
External image preview

### DISCUSSION THREAD

**Discussion by raxon** [directus/directus#12634] [Feature Requests]
Currently you can only use Directus files for image preview. I have problem where I supply images to Directus as URLs, such as some server icon or user avatar.
So maybe it's a good idea to make display as image-external-display which will take [src] for the image address?

**Comment by rijkvanzanten** [comment]
There's a couple gotchas we have to think through: 

- What happens if the referenced URL is auth protected?
- What happens if the referenced URL doesn't have the correct CORS configured, or otherwise prevents previewing?
- Are there any other security/privacy implications to consider?

**Reply by raxon** [reply]
> 
On load we can check type of image jpeg | jpg | gif | png, then we can try load image by JS then if on load is ok on error is something wrong then show default image, also we can verify content-type so:

1. We display default image because image was not loaded,
2. like above 
3. All described on top of post

**Reply by cbertelli** [reply]
I already have images pre-loaded on a file server and I can preview them via an image-CDN, so this is a similar case. I would like to avoid having to upload all images locally. So I just have a path and I can build an URL based on this. Could it be viable as an image field? Or is this a case for a webhook?

**Reply by rijkvanzanten** [reply]
I think a custom interface would be the most appropriate way to go. How would you imagine people entering what file to use? I'd imagine you'd want some sort of UI to browse the existing images and allow them to select it? Or is it purely for presentation purposes and it's a manually added path string?

**Reply by fcdaohang** [reply]
yes. I'm really looking forward to enabling this feature. because I have a big pics collection website and use other CMS to manage it. I want change to Directus. but the images... is a very big headache...
That's very easy If can use external link(URL). just need to use mysql export and import. and in directus's UI can use image preview with external link instead file folder. 

Please. it's very useful for many websitemaster like me.

**Reply by qddtdlko** [reply]
yes. I need this feather (function?) too. a lot of pics cant use upload button to solve.
![image](https://user-images.githubusercontent.com/81072363/165484123-88c23143-d431-459d-bc05-bcc6d23576f8.png)
Please see this pic. if can do that. tks a lot. many webmaster dont know coder much
and
it will add field like
1. allow pics domain (9.0.0)
2. a checkbox if true then show pic url's field use <img>

**Comment by pearcy** [comment]
It seems that this might have been solved.  Here is an article with the code that creates a UI for Directus to use a CDN.  This example uses Coudinary but an option for others (AWS, Akamai, Cloudinary, etc) would be ideal.  

Directus - accessing files
https://docs.directus.io/reference/files.html#accessing-an-file

Article - solution is half way down
https://medium.com/birdie-ai/how-to-manage-your-content-with-directus-and-cloudinary-ded6ddd9a4b1

**Comment by qddtdlko** [comment]
thanks

pearcy ***@***.***> 于2023年1月9日周一 08:02写道：

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
